src/compiler/chainrules.jl

```diff
@@ -175,23 +175,16 @@ As per [`chain_rrule`](@ref) but with support for kwargs.
 end
 
 
-function ChainRulesCore.rrule_via_ad(config::ZygoteRuleConfig, f, args...; kwargs...)
-    kwf(args...) = f(args...; kwargs...)
-    y, _pb = _pullback(config.context, kwf, args...)
-    pb = _isclosure(f) ? _unwrap_closure ∘ _pb : _pb
-    ad_pullback(Δ) = zygote2differential(pb(wrap_chainrules_output(Δ)), (f, args...))
+function ChainRulesCore.rrule_via_ad(config::ZygoteRuleConfig, f_args...; kwargs...)
+    kwf() = first(f_args)(Base.tail(f_args)...; kwargs...)
+    y, pb = _pullback(config.context, kwf)
+    function ad_pullback(Δ)
+        pbres = only(pb(wrap_chainrules_output(Δ))).f_args
```

Comment:
```suggestion
        # Extract tuple of tangents for f_args. (leave the ones for other closed field kwargs behind)
        pbres = only(pb(wrap_chainrules_output(Δ))).f_args
```

`only` was introduced in julia 1.4, but I guess it works here because of having Compat.jl loaded.
(ChainRulesCore loads that)